File: Maximum_Subarray(53).java
/* Given an integer array nums, find the subarray with the largest sum, and return its sum. */

class Solution {
    public int maxSubArray(int[] nums) {
        int len=nums.length;
        int max=Integer.MIN_VALUE;
        int sum=0;
        for(int i=0; i<len; i++){
            sum+=nums[i];

            if(sum>max){
                max=sum;
            }

            if(sum<0){
                sum=0;
            }
        }
        return max;
    }
}

File: Maximum_Subarray(53).cpp
/* Given an integer array nums, find the subarray with the largest sum, and return its sum. */

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n=nums.size();
        int maximum=INT_MIN;
        int sum=0;
        for(int i=0; i<n; i++){
            sum+=nums[i];

            if(sum>maximum){
                maximum=sum;
            }

            if(sum<0){
                sum=0;
            }
        }
        return maximum;
    }
};
